pkg/taskrunmetrics/metrics_test.go

```diff
@@ -585,37 +581,41 @@ func TestRecordCloudEvents(t *testing.T) {
 		expectedCount: 2,
 	}, {
 		name: "for failed task",
-		taskRun: &v1beta1.TaskRun{
+		taskRun: &v1.TaskRun{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      "taskrun-1",
 				Namespace: "ns",
 				Labels: map[string]string{
 					pipeline.PipelineLabelKey:    "pipeline-1",
 					pipeline.PipelineRunLabelKey: "pipelinerun-1",
 				},
+				Annotations: map[string]string{
+					"tekton.dev/v1beta1CloudEvents": `[{"target":"http://event_target","status":{"condition":"Failed","message":"","retryCount":2}}]`,
+				},
 			},
-			Spec: v1beta1.TaskRunSpec{
-				TaskRef: &v1beta1.TaskRef{
+			Spec: v1.TaskRunSpec{
+				TaskRef: &v1.TaskRef{
 					Name: "task-1",
 				},
 			},
-			Status: v1beta1.TaskRunStatus{
+			Status: v1.TaskRunStatus{
 				Status: duckv1.Status{
 					Conditions: duckv1.Conditions{apis.Condition{
 						Type:   apis.ConditionSucceeded,
 						Status: corev1.ConditionFalse,
 					}},
 				},
-				TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+				TaskRunStatusFields: v1.TaskRunStatusFields{
 					StartTime:      &metav1.Time{Time: time.Now()},
 					CompletionTime: &metav1.Time{Time: time.Now().Add(1 * time.Minute)},
-					CloudEvents: []v1beta1.CloudEventDelivery{{
-						Target: "http://event_target",
-						Status: v1beta1.CloudEventDeliveryState{
-							Condition:  v1beta1.CloudEventConditionFailed,
-							RetryCount: 2,
-						},
-					}},
+					// TODO, use conversion test to rewrite
```

Comment:
it looks like you might have forgotten to address this TODO (and the other one in this file)